src/lib/Cashlink.ts

```diff
@@ -0,0 +1,453 @@
+        return Utf8Tools.utf8ByteArrayToString(this._messageBytes);
+    }
+
+    set message(message) {
+        if (this._messageBytes.byteLength && (this._immutable || this.state !== CashlinkState.UNCHARGED)) {
+            throw new Error('Cashlink is immutable');
+        }
+        const messageBytes = Utf8Tools.stringToUtf8ByteArray(message);
+        if (!Nimiq.NumberUtils.isUint8(messageBytes.byteLength)) throw new Error('Message is too long');
+        this._messageBytes = messageBytes;
+    }
+
+    set networkClient(client: NetworkClient) {
+        this._networkClientResolver(client);
+    }
+
+    public static async create(): Promise<Cashlink> {
+        await loadNimiq();
+        const keyPair = Nimiq.KeyPair.derive(Nimiq.PrivateKey.generate());
+        return new Cashlink(keyPair, keyPair.publicKey.toAddress());
+    }
+
+    public static async parse(str: string): Promise<Cashlink | null> {
+        if (!str) return null;
+        try {
+            str = str.replace(/~/g, '').replace(/=*$/, (match) => new Array(match.length).fill('.').join(''));
+            const buf = Nimiq.BufferUtils.fromBase64Url(str);
+            await loadNimiq();
+            const keyPair = Nimiq.KeyPair.derive(Nimiq.PrivateKey.unserialize(buf));
+            const value = buf.readUint64();
+            let message;
+            if (buf.readPos === buf.byteLength) {
+                message = '';
+            } else {
+                const messageLength = buf.readUint8();
+                const messageBytes = buf.read(messageLength);
+                message = Utf8Tools.utf8ByteArrayToString(messageBytes);
+            }
+
+            return new Cashlink(
+                keyPair,
+                keyPair.publicKey.toAddress(),
+                value,
+                message,
+                CashlinkState.UNKNOWN);
+        } catch (e) {
+            console.error('Error parsing Cashlink:', e);
+            return null;
+        }
+    }
+
+    public static fromObject(object: CashlinkEntry): Cashlink {
+        return new Cashlink(
+            Nimiq.KeyPair.unserialize(new Nimiq.SerialBuffer(object.keyPair)),
+            Nimiq.Address.fromUserFriendlyAddress(object.address),
+            object.value,
+            object.message,
+            object.state,
+            object.date,
+            object.contactName,
+        );
+    }
+
+    private _network: Promise<NetworkClient>;
+    private _networkClientResolver!: (client: NetworkClient) => void;
+    private _balanceRequest: Promise<number> | null = null;
+    private _currentBalance: number = 0;
+    private _immutable: boolean;
+    private _eventListeners: {[type: string]: Array<(data: any) => void>} = {};
+    private _messageBytes: Uint8Array = new Uint8Array(0);
+    private _value: number | null = null;
+    private _knownTransactions: DetailedPlainTransaction[] = [];
+
+    constructor(
+        public keyPair: Nimiq.KeyPair,
+        public address: Nimiq.Address,
```

Comment:
This is done like this to not have to load the Nimiq lib when managing already-stored Cashlinks. Because WASM is not required (in the Hub) for monitoring the blockchain and listening for txs, the Nimiq lib is only needed for
* creation (generating a private key and deriving the keypair & address)
* parsing (deriving keypair & address)
* claiming (signing a transaction)

Just loading a Cashlink from storage and displaying it's link or monitoring it thus does not require the Nimiq lib.

Reply:
Note that getting the userfriendly address from the keypair does _not_ require WASM.